src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -619,6 +620,28 @@ A list of all generated pages. It is an object with one property.
 
 - `pathname` - the finalized path of the page.
 
+### `astro:route:setup`
+
+<p><Since v="4.14.0" /></p>
+
+**When:** In `astro dev`, whenever a route is requested. In `astro build`, while bundling and transforming a route file.
+
+**Why:** To update a route's option, i.e. `prerender` to configure [on-demand server rendering](/en/guides/server-side-rendering/#opting-in-to-pre-rendering-in-server-mode).
+
+```js
+'astro:route:setup'?: (options: { route: RouteOptions }) => void | Promise<void>;
+```
+
+#### `route` option
+
+**Type:** `RouteOptions`
+
+An object with a readonly `component` property and a `prerender` property.
```

Comment:
```suggestion
An object with a `component` property to identify the route and the following additional values to allow you to configure the generated route: `prerender`.
```
So, thinking future-proofing here, it would be nice to set this up for not having to *totally* reconstruct the sentence as you add more of these. Maybe something like that?

I know it feels a little artificial when there's only *one* configurable value, but I like the idea of something that sets us up for: identify the component, and then here's all the updates you can make to this file when actually creating the route.

Reply:
Sounds good to me. I didn't word it that way because I don't see us probably adding a new field in a looong time, but I could be wrong 😄 But anyways I don't think it hurts to futureproof it now